### PR TITLE
Localist scraper/parser

### PR DESCRIPTION
**Pull request by Enochen**
### Scraper 

Adds the ability to scrape event data from sites using the Localist api and import events into the database.

##### Scraping and saving data to file
First, use the `localist_fetch.py` script to download the data into a json format.
Change the parameters within the script (may be changed to CLI args later) to determine the range of dates to scrape.
The script will save the json in its folder as event-{start_date}-{end_date}.json.

##### Adding tags
Then, you want to manually add tags to each event.
Within the json structure, each tags will have an empty list called tags.
Populate these - for example, `tags = ["Free-Food", "Other", "A&S"]`
Tags themselves can be added by running `py .\events_backend\manage.py add_tags` from root.

##### Google Maps place_id
(Optional) You can run the `inject_placeid.py` script to insert appropriate place_id values in json.
Make sure to edit the parameters inside the script, especially the Google Maps API key.

##### Adding to database
Finally, you want to import the json into the database.
To do this, run `py .\events_backend\manage.py localist_parser [path-to-json]`.
If you have a csv (from facebook instead of localist), run `py .\events_backend\manage.py csv_event_parser [path-to-csv]` instead.

### Notes 

Right now the scraped img_src urls are stored directly, but they are not guaranteed to be stable.
We can fix this later by uploading a copy of the images to S3 and point to the S3 url instead.

Attendance numbers are not provided by Localist, so they are at a default of 0.
If missing from data, the default email is `donotdisplay@cornell.edu`

### CODE REVIEW

File: events_backend/app/management/commands/add_tags.py [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+from dateutil.parser import parse
+from django.core.management.base import BaseCommand, CommandError
+
+
+class Command(BaseCommand):
+    help = 'Parses events from csv and add them to the database'
+
+    # def add_arguments(self, parser):
+    #     parser.add_argument(
+    #         'path', help='Indicates the path of json to be parsed')
```
Comment (SamChou19815) [inline events_backend/app/management/commands/add_tags.py]: Do not keep commented code.

File: misc/scraper/localist_fetch.py [hunk 2]
```diff
@@ -0,0 +1,122 @@
+import json
+
+import requests
+
+if __name__ == '__main__':
+    MAX_EVENT_PER_PAGE = 100
+    SCRAPER_START_DATE = "2019-10-30"
+    SCRAPER_END_DATE = "2019-10-31"
+
+    URL = "https://events.cornell.edu/api/2/events?start={0}&end={1}&pp={2}".format(SCRAPER_START_DATE,
```
Comment (SamChou19815) [inline misc/scraper/localist_fetch.py]: Nits: Using python3's format string can make it more readable:
```python
f"https://events.cornell.edu/api/2/events?start={SCRAPER_END_DATE}&end={SCRAPER_START_DATE}&pp={MAX_EVENT_PER_PAGE}"
```

File: misc/scraper/localist_fetch.py [hunk 3]
```diff
@@ -0,0 +1,122 @@
...
+                    continue
+
+                if org_name == "" or org_name is None:
+                    print("Missing organizer name")
+                    data_count += 1
+                    fail_data_count += 1
+                    continue
+
+                if (contact_email == "" or contact_email is None):
+                    # print("Missing organizer email")
```
Comment (SamChou19815) [inline misc/scraper/localist_fetch.py]: 🤔

File: misc/scraper/localist_fetch.py (hunk 2 above)
Comment (Enochen) [reply misc/scraper/localist_fetch.py]: Ooh thanks for the heads up!

**Comment by abagh0703** [comment]
@Enochen  want to merge in?

**Pull request merged**